### PR TITLE
updating consumer and producer with request write changes

### PR DESCRIPTION
**Pull request by beemarie**

### CODE REVIEW

**Comment by beemarie** [comment]
Closes #35

**Comment by beemarie** [comment]
@vagababov I think this should address your feedback from here:
https://github.com/knative-sandbox/async-component/pull/27#discussion_r517529042

File: cmd/consumer/main.go [hunk 1]
```diff
@@ -46,23 +46,14 @@ func consumeEvent(event cloudevents.Event) error {
...
-		return fmt.Errorf("problem reading request: %w", err)
-	}
 	// client for sending request
 	client := &http.Client{}
-
-	// build new url - writing the request removes the URL and places in URI.
-	req.URL, err = url.Parse("http://" + req.Host + req.RequestURI)
-	if err != nil {
-		return fmt.Errorf("Problem parsing url: %w", err)
+	req, err := http.NewRequest(data.ReqMethod, data.ReqURL, nil)
```
Comment (maximilien) [inline cmd/consumer/main.go:51]: What happens if there is an error when creating a `NewRequest`?

**Review by maximilien — APPROVED** [review]
As more a convention than anything, I typically would keep `main` small and not unit tested and move the main logic to other files which would then be tested. 

Of course, `e2e` and integration tests would be testing the whole thing and thus the main entry point.

Anyhow, probably best follow the current convention in serving as you see fit.

File: cmd/consumer/main.go (hunk 1 above)
Comment (beemarie) [reply cmd/consumer/main.go:51]: just updated with an error check, thanks.

**Comment by beemarie** [comment]
> As more a convention than anything, I typically would keep `main` small and not unit tested and move the main logic to other files which would then be tested.
> 
> Of course, `e2e` and integration tests would be testing the whole thing and thus the main entry point.
> 
> Anyhow, probably best follow the current convention in serving as you see fit.

Looking through the serving components, there are functional main files in activator and autoscaler at least. I do think it's reasonable to ensure that we're following go conventions & knative conventions for the project structure. I just created an issue to take a look at that:
https://github.com/knative-sandbox/async-component/issues/45

**Review by maximilien — APPROVED** [review]
/lgtm

**Pull request merged**